lsp-serenata.el

```diff
@@ -0,0 +1,61 @@
+;;; lsp-serenata.el --- Serenata server configuration         -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2020 Fermin Munoz
+
+;; Author: Fermin Munoz <fmfs@posteo.net>
+;; Keywords: php lsp
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; lsp-serenata client
+
+;;; Code:
+
+(require 'lsp-mode)
+
+(defgroup lsp-serenata nil
+  "LSP support for the PHP programming language, using serenata"
+  :group 'lsp-mode
+  :link '(url-link "https://gitlab.com/Serenata/Serenata"))
+
+
+(defcustom lsp-serenata-server-path
+  "serenata.phar"
+  "Path to the Serenata Language Server phar file.
+It can be downloaded from https://gitlab.com/Serenata/Serenata/-/releases."
+  :group 'lsp-serenata
+  :type 'file)
+
+
+(defun lsp-serenata-server-start-fun (port)
+  "Define serenata start function, it requires a PORT."
+  `(, lsp-serenata-server-path
+    "-u" ,(number-to-string port)))
+
+
+(lsp-register-client
+ (make-lsp-client
+  :new-connection (lsp-tcp-connection 'lsp-serenata-server-start-fun)
+  :major-modes '(php-mode)
+  :priority -1
```

Comment:
the priority should not be the same as the other php servers - how do you compare seranata to the others?

Reply:
We should really group all `php` related language server into one file, it's also easier to compare the priority too

Reply:
> We should really group all `php` related language server into one file, it's also easier to compare the priority too

Let's keep this as part of the different PR and eventually decide if we want to follow that. Splitting into multiple files as advantages too - e. g. you might skip loading the file via lsp-client-packages.

Reply:
> the priority should not be the same as the other php servers - how do you compare seranata to the others?
Serenata is less powerfull than intelphense or php-language-server, it still doesn't have find-references or workspace-symbols.
 It is the less mature of the 3, so maybe the lowest priority